Build fails with strict-aliasing violations

### Description

```
FAILED: tests/CMakeFiles/test_pfx.dir/test_pfx.c.o 
/usr/bin/x86_64-pc-linux-gnu-gcc  -I/var/tmp/portage/net-libs/rtrlib-0.8.0/work/rtrlib-0.8.0/.  -march=native -fstack-protector-all -O2 -pipe -fdiagnostics-color=always -frecord-gcc-switches -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-clash-protection -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing  -Wformat -Werror=format-security -Werror=implicit-function-declaration -Werror=implicit-int -Werror=int-conversion -Werror=incompatible-pointer-types -Wall -std=gnu99 -fstack-protector-all -MD -MT tests/CMakeFiles/test_pfx.dir/test_pfx.c.o -MF tests/CMakeFiles/test_pfx.dir/test_pfx.c.o.d -o tests/CMakeFiles/test_pfx.dir/test_pfx.c.o -c /var/tmp/portage/net-libs/rtrlib-0.8.0/work/rtrlib-0.8.0/tests/test_pfx.c
/var/tmp/portage/net-libs/rtrlib-0.8.0/work/rtrlib-0.8.0/tests/test_pfx.c: In function ‘mass_test’:
/var/tmp/portage/net-libs/rtrlib-0.8.0/work/rtrlib-0.8.0/tests/test_pfx.c:132:54: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
  132 |                 ((uint64_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
      |                                                      ^
/var/tmp/portage/net-libs/rtrlib-0.8.0/work/rtrlib-0.8.0/tests/test_pfx.c:152:54: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
  152 |                 ((uint64_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
      |                                                      ^
/var/tmp/portage/net-libs/rtrlib-0.8.0/work/rtrlib-0.8.0/tests/test_pfx.c:176:54: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
  176 |                 ((uint64_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
      |                                                      ^
cc1: some warnings being treated as errors
```


### Steps to reproduce the issue

I tried to compile with the following flags for LTO: `-flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing`

The -Werror's here are very important, because LTO can optimize based on the assumption that certain language requirements are met, and if they aren't met, a miscompilation happens.

Actually, strict-aliasing can cause issues even without LTO ;) but LTO does tend to make them worse.

Downstream report: https://bugs.gentoo.org/861581
Full build logs: [build.log](https://github.com/rtrlib/rtrlib/files/14490283/build.log)


### Versions
Operating system: Gentoo Linux
Build environment: GCC
rtrlib: 0.8.0

--- a/tests/test_pfx.c
+++ b/tests/test_pfx.c
@@ -128,8 +128,8 @@ static void mass_test(void)
 		pfx.min_len = 128;
 		pfx.max_len = 128;
 		pfx.prefix.ver = LRTR_IPV6;
-		((uint64_t *)pfx.prefix.u.addr6.addr)[1] = max_i;
-		((uint64_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
+		((uint32_t *)pfx.prefix.u.addr6.addr)[2] = max_i;
+		((uint32_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
 		assert(pfx_table_add(&pfxt, &pfx) == PFX_SUCCESS);
 	}
 
@@ -148,8 +148,8 @@ static void mass_test(void)
 		pfx.min_len = 128;
 		pfx.max_len = 128;
 		pfx.prefix.ver = LRTR_IPV6;
-		((uint64_t *)pfx.prefix.u.addr6.addr)[1] = max_i;
-		((uint64_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
+		((uint32_t *)pfx.prefix.u.addr6.addr)[2] = max_i;
+		((uint32_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
 
 		assert(pfx_table_validate(&pfxt, i + 1, &pfx.prefix, pfx.min_len, &res) == PFX_SUCCESS);
 		assert(res == BGP_PFXV_STATE_VALID);
@@ -172,8 +172,8 @@ static void mass_test(void)
 		pfx.prefix.ver = LRTR_IPV6;
 		pfx.min_len = 128;
 		pfx.max_len = 128;
-		((uint64_t *)pfx.prefix.u.addr6.addr)[1] = max_i;
-		((uint64_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
+		((uint32_t *)pfx.prefix.u.addr6.addr)[2] = max_i;
+		((uint32_t *)pfx.prefix.u.addr6.addr)[0] = min_i + i;
 		assert(pfx_table_remove(&pfxt, &pfx) == PFX_SUCCESS);
 	}
 

